### DISCUSSION TITLE
'npx supabase gen types' makes timestampz to string.

### DISCUSSION THREAD

**Discussion by tetsuyanh** [supabase/supabase#20971] [Questions] [answered]
Do we have to manage type definitions manually or parse date?

**Comment by mwoss** [comment, marked as answer]
As far as I know, all dates are returned as strings by default as there is no `date` type in JSON specification and Supabase uses PostgREST underneath for communication with clients. Supabase crew is thinking about adding automatic date conversion support in suopbase-js v3 tho - https://github.com/supabase/cli/issues/467#issuecomment-1253508524

**Reply by tetsuyanh** [reply]
Thank you for your prompt response. I understand well now.
